filewatch/fwork_windows.go

```diff
@@ -30,22 +30,32 @@ func getFileId(f *os.File) (id FileId, err error) {
 	return
 }
 
+// getFileIdFromName is a windows version of stat that returns something roughly equal to a device and inode.
+// there is voodoo, we all hate it... yet here we are...
 func getFileIdFromName(name string) (id FileId, err error) {
-
+	var attrib *syscall.SecurityAttributes
 	p, lerr := syscall.UTF16PtrFromString(name)
 	if lerr != nil {
 		err = lerr
 		return
 	}
-	h, lerr := syscall.CreateFile(p, 0, 0, nil, syscall.OPEN_EXISTING, syscall.FILE_FLAG_BACKUP_SEMANTICS, 0)
+
+	//make sure to drop the share attributes even in this call because apparently
+	//GetFileINformationByHandle syscall can take time... It looks like a stat...
+	//it smells like a stat... it ain't a stat....
+	shared := uint32(syscall.FILE_SHARE_READ | syscall.FILE_SHARE_WRITE | syscall.FILE_SHARE_DELETE)
+	h, lerr := syscall.CreateFile(p, syscall.GENERIC_READ, shared, attrib, syscall.OPEN_EXISTING, syscall.FILE_FLAG_BACKUP_SEMANTICS, 0)
 	if lerr != nil {
-		err = fmt.Errorf("Failed to open %s to check FileID: %v", name, lerr)
+		err = fmt.Errorf("Failed to open %s to check FileID: %w", name, lerr)
 		return
 	}
 	defer syscall.CloseHandle(h)
 	var bhfi syscall.ByHandleFileInformation
 	if err = syscall.GetFileInformationByHandle(h, &bhfi); err != nil {
-		err = fmt.Errorf("Failed to get FileID for %s: %v", name, err)
+		if os.IsNotExist(err) {
+			err = os.ErrNotExist
```

Comment:
This is extraneous given the way you re-declare err on line 58

Reply:
the problem is that for some damn reason after i wrap that error in path error it wasn't tripping.

The standard library does something similar in windows, where it checks the error, and if its X or Y return the constant stdlib error.  I am doing something similar.

Reply:
cleaning it up a bit though